src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquirerFactory.cs

```diff
@@ -29,14 +29,33 @@ public class TokenAcquirerFactory : ITokenAcquirerFactory
         /// <summary>
         /// Services. Used in the initialization phase.
         /// </summary>
-        public ServiceCollection Services { get; protected set; } = new ServiceCollection();
+        /// <exception cref="InvalidOperationException"/> will be thrown if you try to access
+        /// Services after you called <see cref="Build"/>.
+        public ServiceCollection Services
+        {
+            get
+            {
+                if (ServiceProvider != null)
+                {
+                    throw new InvalidOperationException("Cannot change services once you called Build()");
```

Comment:
We want to prevent partners to access services once the DI resolution has happened (to avoid mistakes)